### ISSUE TITLE
Failed to minify the code from this file: react-lifecycles-compat.es.js:64

### ISSUE THREAD

**Issue by joaom182** [reactjs/react-lifecycles-compat#25]
Hello,

I'm using `react-scripts` to generate a build production to deploy my application, and on the build step throws this error:

```
Failed to minify the code from this file:
        ./node_modules/react-lifecycles-compat/react-lifecycles-compat.es.js:64
```

Have any chance to help investigate what in this file might be causing the minification failure?

As a hot fix i've minified manually the file on https://babeljs.io/repl

**Comment by bvaughn** [comment]
Can you share your repro case?

**Comment by joaom182** [comment]
I have any lib thats depends of `react-lifecycles-compat`, my application not depends directly of `react-lifecycles-compat`, but on dependency graph, the `create-react-app/react-scripts` try to minify the `react-lifecycles-compat` and throws the error.

## Sugestion to reproduce ##
Create a blank react-app using `create-react-app` --> https://github.com/facebook/create-react-app/blob/master/README.md#getting-started, install `react-lifecycles-compat` as dependency and import in any file and try build the application with `react-scripts build`

## Sugestion of `create-react-app` ##

![image](https://user-images.githubusercontent.com/3612098/39935749-69db45f2-5520-11e8-8608-afa50209fbde.png)

**Comment by bvaughn** [comment]
To further clarify, @joaom182, which version of `react-lifecycles-compat` are you using? The error you've provided points to line 64, which is just a comment in the latest couple of releases.

Also, following the repro steps you listed above, I am unable to reproduce this error.

1. Create a new app using `create-react-app`
2. Install `react-lifecycles-compat`
3. Add a component that uses the `react-lifecycles-compat` polyfill
4. Run `yarn build` to create a production bundle
5. Test production bundle by running `serve -s build` and opening localhost:5000

**Comment by joaom182** [comment]
`3.0.0`

**Comment by joaom182** [comment]
My project depends of `react-form-validator-core@0.4.3` that depends of `react-lifecycles-compat@3.0.0`

**Comment by bvaughn** [comment]
Thanks for clarifying!

3.0.0 had some newer syntax (`let` and `const` keyword and a template literal) that mistakenly slipped in. Versions 3.0.0 and 3.0.1 both had this problem , so the fix is to update to 3.0.2+.

(This is mentioned in [the CHANGELOG](https://github.com/reactjs/react-lifecycles-compat/blob/master/CHANGELOG.md) if you're curious.)

I've marked both versions 3.0.0 and 3.0.1 as deprecated so you should see the following warning when installing:
```
$  npm i react-lifecycles-compat@3.0.0
npm WARN deprecated react-lifecycles-compat@3.0.0: Contains accidental untrasnpiled ES6 syntax. Prefer 3.0.2+
$ npm i react-lifecycles-compat@3.0.1
npm WARN deprecated react-lifecycles-compat@3.0.1: Contains accidental untrasnpiled ES6 syntax. Prefer 3.0.2+
```

For now, the appropriate fix would be to file an issue with `react-form-validator-core`. Sorry for the inconvenience!
